src/include/rbd/librbd.h

```diff
@@ -773,6 +773,10 @@ CEPH_RBD_API int rbd_list_children3(rbd_image_t image,
                                     rbd_linked_image_spec_t *images,
                                     size_t *max_images);
 
+CEPH_RBD_API int rbd_list_descendants(rbd_image_t image,
```

Comment:
Nit: just thinking aloud, but I wonder if this interface should link a parent image spec with the child image spec in the results? If you have a decedent that is several layers deep, you could then easily understand the hierarchy of the layering without the need to re-open the images.

Reply:
There are many cases (like this one) when you don't need to know hierarchy, but just to know if there any descendants, and if there are then what they are. And in such cases the function like this with a simple interface fits well. On the other hand providing both parent and child would be more generic and most likely useful for other cases.

I don't mind and can do this way, and the only reason I did not do this  initially that the function signature will be much more complicated. Do you have an idea about the interface? How we could return this info? For `c++`, it could be a list or vector of parent-child pairs, and for `c` it could be two vectors: `parents` and `children`. I suppose we don't want something more complicated like `parent` -> `children` map?

Reply:
I was thinking more about future-proofing, but I guess if and when the need ever comes, we can just add a version 2 API method.